uber/automated_emails.py

```diff
@@ -96,19 +141,72 @@ def __init__(self, conf):
         AutomatedEmail.__init__(self, Attendee,
                                 subject='{EVENT_NAME} Department Checklist: ' + conf.name,
                                 template='shifts/dept_checklist.txt',
-                                filter=lambda a: a.is_single_dept_head and a.admin_account and days_before(7, conf.deadline) and not conf.completed(a),
+                                filter=lambda a: a.is_single_dept_head and a.admin_account and not conf.completed(a),
+                                date_filters=days_before(7, conf.deadline),
                                 sender=c.STAFF_EMAIL,
                                 extra_data={'conf': conf})
 
-before = lambda dt: bool(dt) and localized_now() < dt
-after = lambda dt: bool(dt) and localized_now() > dt
-days_after = lambda days, dt: bool(dt) and (localized_now() > dt + timedelta(days=days))
 
+print_dateformat = "%m/%d"
+
+
+class days_before:
+    def __init__(self, days, dt, until=None):
+        self.dt = dt
+        self.days = days
+        self.until = until
+
+        if dt:
+            self.starting_date = self.dt - timedelta(days=self.days)
+            self.ending_date = (dt - timedelta(days=until)) if until else dt
+
+    def __call__(self):
+        return self.starting_date < localized_now() < self.ending_date if self.dt else False
+
+    @property
+    def active_when(self):
+        return 'between {} and {}'.format(self.starting_date.strftime(print_dateformat),
+                                          self.ending_date.strftime(print_dateformat)) \
+            if self.dt else ''
+
+
+class days_after:
+    def __init__(self, days, dt):
+        self.dt = dt
+        self.days = days
+
+        self.starting_date = dt + timedelta(days=days) if dt else None
+
+    def __call__(self):
+        return bool(self.dt) and (localized_now() > self.starting_date)
+
+    @property
+    def active_when(self):
+        return 'after {}'.format(self.starting_date.strftime(print_dateformat)) if self.starting_date else ''
```

Comment:
This is definitely a touch call on where and how to draw the line on presentation code being separate from business logic code.  FWIW, I'm generally okay with things having methods/properties which return textual labels like this.  The point where I'd draw the line and be against it is if we were returning HTML or CSS or something.  But for me at least, a label like this is an okay compromise.  (Consider that it might go into both an HTML page as well as a CSV export, etc.)